packages/builtin/array.pony

```diff
@@ -1,6 +1,139 @@
 class Array[A] is Seq[A]
   """
   Contiguous, resizable memory to store elements of type A.
+
+  # Usage
+  Creating an array of Strings, single line.
+  ```pony
+    let array: Array[String] = ["dog"; "cat"; "wombat"]
+  ```
+
+  Semicolon is an expression, so this is a multiline equivalent.
+   ```pony
+    let array: Array[String] = [
+      "dog"
+      "cat"
+      "wombat"
+    ]
+  ```
+
+  Accessing elements can be done via the `apply(i: USize): this->A ?` method.
+  The index provided might be out of bounds so `apply` is partial and has to be
+  called within a try-catch block or inside another partial method.
+  ```pony
+    fun second_element_is_wombat(array: Array[String]): Bool =>
+      try
+        // first element's index is 0, so second is 1
+        return array(1)? == "wombat"
+      else
+        return false
+      end
+
+      // the second element is indeed a wombat
+      second_element_is_wombat(["star"; "wombat"]) == true
+  ```
+
+  # Memory allocation
+  Array allocates contiguous memory. It always allocates at least enough memory
+  space to hold all of it's elements. Space is the number of elements the array
+  can hold without allocating more memory. The exact amout of memory needed for
+  one unit of space depends on the type of elements the array can hold.
```

Comment:
I believe the idea here is that an Array[U64] with size of 6 will take more memory than an Array[U8] of the same size.

I think that point could be made more clear with some a couple of concrete examples.